torchtext/experimental/transforms.py

```diff
@@ -174,6 +175,143 @@ def to_ivalue(self):
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_tokenizer = sentencepiece_tokenizer(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_tokenizer)
+    """
+    return SentencePieceTokenizer(spm_model)
+
+
+class SentencePieceTokenizer(nn.Module):
+    r"""Tokenizer based on a pretained sentencepiece model.
+
+    Args:
+       spm_model: the sentencepiece model instance
+    """
+
+    def __init__(self, spm_model):
+        super(SentencePieceTokenizer, self).__init__()
+        self.sp_model = spm_model
+
+    def forward(self, line: str) -> List[str]:
+        r"""
+        Args:
+            line: the input sentence string
+
+        Examples:
+            >>> spm_tokenizer('the pretrained sp model names')
+            >>> ['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names']
+        """
+        return self.sp_model.EncodeAsPieces(line)
+
+    @torch.jit.export
+    def decode(self, tokens: List[str]) -> str:
+        r"""
+        Args:
+            tokens: the tokens list for decoder
+
+        Examples:
+            >>> spm_transform.decoder(['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names'])
+            >>> 'the pretrained sp model names'
+        """
+        return self.sp_model.DecodePieces(tokens)
+
+
+def sentencepiece_transform(spm_model):
+    r"""factory function to generate SentencePieceTransform with pretrained sp model
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTransform
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_transform = sentencepiece_transform(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_transform)
+    """
+    return SentencePieceTransform(spm_model)
+
+
+class SentencePieceTransform(nn.Module):
+    r"""String to ids transform based on a pretained sentencepiece model
+
+    Args:
+       spm_model: the sentencepiece model instance
+    """
+
+    def __init__(self, spm_model):
+        super(SentencePieceTransform, self).__init__()
+        self.sp_model = spm_model
+
+    def forward(self, line: str) -> List[int]:
+        r"""
+        Args:
+            line: the input sentence string
+
+        Examples:
+            >>> spm_transform('the pretrained sp model names')
+            >>> [9, 1546, 18811, 2849, 2759, 2202]
+        """
+        return self.sp_model.EncodeAsIds(line)
+
+    @torch.jit.export
+    def decode(self, ids: List[int]) -> str:
+        r"""
+        Args:
+            ids: the integers list for decoder
+
+        Examples:
+            >>> spm_transform.decoder([9, 1546, 18811, 2849, 2759, 2202])
+            >>> 'the pretrained sp model names'
+        """
+        return self.sp_model.DecodeIds(ids)
+
+
 class VocabTransform(nn.Module):
```

Comment:
Unrelated: do we want / need the `Transform` name here? Isn't everything in this file a transform anyway?